resolver/type_syntax.cc

```diff
@@ -558,7 +574,15 @@ TypeSyntax::ResultType interpretTCombinator(core::Context ctx, const ast::Send &
             auto *obj = ast::cast_tree<ast::ConstantLit>(send.args[0]);
             if (!obj) {
                 if (auto e = ctx.beginError(send.loc, core::errors::Resolver::InvalidTypeDeclaration)) {
-                    e.setHeader("T.class_of needs a Class as its argument");
+                    auto type = getResultTypeWithSelfTypeParams(ctx, send.args.front(), sig, args);
+                    auto autocorrect = autoCorrectOrOfOnlyClasses(ctx, type);
+                    if (core::isa_type<core::OrType>(type) && autocorrect) {
+                        e.setHeader("`{}` needs a class or module as its argument. Apply `{}` to each element of the `{}` instead",
+                                    "T.class_of", "T.class_of", "T.any");
```

Comment:
I'm not a fan of double sentences in the header for the same reason as the "no period at the end" ENFORCE.

What do you think about just mentioning the more relevant message?

```suggestion
                        e.setHeader("`{}` must wrap each individual class type, not the outer `{}`", "T.class_of", "T.any");
```